### PR TITLE
Framework: Fix logged-out /design render with persistence enabled

### PR DESCRIPTION
**Pull request by seear**
**Before**
screen shot 2016-02-08 at 17 59 46

**After**
screen shot 2016-02-08 at 17 52 26

**To Test**
1) Set `persist-redux` to `true` in `config/development.json`
2) Go to http://calypso.localhost:3000/design and check there is no sidebar-sized gap on the left

When the `persist-redux` flag is enabled, redux state bootstrapped from the server (on the logged-out /design route) is passed through the `DESERIALIZE` actions. This change preserves the state passed through those actions for `section` and `hasSidebar`, which are the two fields needed for the client render of logged-out /design.

/cc @gwwar 

An alternative to tweaking the `DESERIALIZE` actions would be to merge in the bootstrapped state after deserialization. The downside to that approach being that we would not be taking advantage of the immutable conversions.

### CODE REVIEW

File: client/state/ui/reducer.js [hunk 1]
```diff
@@ -79,7 +79,7 @@ export function hasSidebar( state = true, action ) {
 		case SERIALIZE:
 			return true;
 		case DESERIALIZE:
-			return true;
+			return state;
```
Comment (gwwar) [inline client/state/ui/reducer.js]: Don't forget to update the related tests :)

File: client/state/ui/reducer.js [hunk 2]
```diff
@@ -79,7 +79,7 @@ export function hasSidebar( state = true, action ) {
 		case SERIALIZE:
 			return true;
```
Comment (gwwar) [inline client/state/ui/reducer.js]: I'm still working on a good validation strategy in #3101, but if it makes sense to persist state, feel free to serialize it as well. So perhaps ``` return state; ``` here and in `section`

**Comment by gwwar** [comment]
:+1: feel free to merge after you update tests in `client/state/ui`

**Comment by seear** [comment]
@gwwar: thanks for review :)

File: client/state/ui/test/reducer.js [hunk 3]
```diff
@@ -78,38 +78,38 @@ describe( 'reducer', () => {
...
 
 	describe( '#section()', () => {
-		it( 'should not persist data because this is not implemented', () => {
+		it( 'should persist data', () => {
 			const state = section( 'hello', {
 				type: SERIALIZE
 			} );
 
-			expect( state ).to.eql( false );
+			expect( state ).to.eql( 'hello' );
```
Comment (gwwar) [inline client/state/ui/test/reducer.js]: :) Might make more sense to change this to a boolean.

Comment (seear) [inline client/state/ui/test/reducer.js]: > Might make more sense to change this to a boolean.

`section` can be a string ('design', 'reader' etc) or `false`, so I think the string is ok, or am I missing something?

Comment (gwwar) [inline client/state/ui/test/reducer.js]: Oh, I forgot we mixed types here! :)

**Pull request merged**